### PR TITLE
FAD-4440 Display DKIM errors

### PR DESCRIPTION
**Pull request by jonambas**
I had to override the Table component through an optional TRComponent prop in order to pull in the signature row error and display it. The table mapping makes it a bit confusing, and we should probably move to something other than a table (with a new design) after launch.

Also removed some spf css classes to use global helpers

### CODE REVIEW

**Comment by jonambas** [comment]
screen shot 2017-02-16 at 1 51 08 pm

File: src/pages/dkim/components/SignatureTable.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Cell } from 'components/table/Table';
+
+const StatusCell = ({ value, error }) => {
+  const type = value === 'Passed' ? 'is-valid' : 'has-error';
+  return (
+    <td className='table__cell'>
+      <span className={`text--semibold ${type}`}>{value}</span>
+      {error && <div className='text--small'>{error}</div>}
```
Comment (colestrode) [inline src/pages/dkim/components/SignatureTable.js]: we probably want some sort of overflow here. I tried a very long error message and it pushes the the content outside of the panel.

For now the longest error message will be `Optional v tag must be first if present in DKIM record` which fits inside the panel, but maybe have an overflow property for safety?

File: src/components/Icon.js [hunk 2]
```diff
@@ -10,4 +10,4 @@ const mapPropsToClasses = ({ name, size, extras = [] }) => (
   }, extras)
 );
 
-export default (props) => <i className={mapPropsToClasses(props)} style={{ transition: 'all 200ms ease-out' }} />;
```
Comment (jonambas) [inline src/components/Icon.js]: Just removed it, transitions were already defined in sass (for autoprefixing) and only in places where we need it

File: src/components/table/Table.js [hunk 3]
```diff
@@ -9,13 +9,18 @@ const Row = ({ values }) => <tr className='table__row'>{values.map((v, i) => <Ce
...
+const Table = ({ headers, rows, TRComponent }) => (
   <table className='table table--transparent'>
     <thead className='table__head'>
       <tr className='table__row'>{headers.map((h, i) => <HeaderCell value={h} key={`header-${i}`} />)}</tr>
     </thead>
     <tbody className='table__body'>
-      {rows.map((row, i) => <Row values={row} key={`row-${i}`} />)}
+      {rows.map((row, i) => {
+        if (TRComponent) {
+          return <TRComponent values={row} key={`row-${i}`} />;
```
Comment (jonambas) [inline src/components/table/Table.js]: Turns out { RowComponent = Row } does work!

File: src/components/table/Table.js [hunk 4]
```diff
@@ -8,19 +8,15 @@ const Row = ({ values }) => <tr className='table__row'>{values.map((v, i) => <Ce
 /**
  * Produces a standard table with a header row
  * and n regular rows below
+ * Accepts an optional Row component
  */
-const Table = ({ headers, rows, TRComponent }) => (
+const Table = ({ headers, rows, RowComponent = Row }) => (
```
Comment (colestrode) [inline src/components/table/Table.js]: nice defaulting

Comment (jonambas) [inline src/components/table/Table.js]: I think I had "**row**Component" before, and jsx wouldn't render a component beginning with a lowercase letter

**Pull request merged**